### PR TITLE
feat(ui): add check_images handlers to the controller slice

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add support for checking images with polling to the controllers slice.
- Add support for multiple polling events for the same endpoint to the websocket client.

## Fixes

Fixes: canonical-web-and-design/app-tribe#611.

### CODE REVIEW

File: ui/src/app/base/sagas/websockets.ts [hunk 1]
```diff
@@ -545,7 +558,8 @@ const buildMessage = (
     method: `${meta.model}.${meta.method}`,
     type: WebSocketMessageType.REQUEST,
   };
-  if (params && !Array.isArray(params)) {
+  const hasMultipleDispatches = meta.dispatchMultiple && Array.isArray(params);
```
Comment (huwshimi) [inline ui/src/app/base/sagas/websockets.ts:561]: Previously we didn't have any websocket endpoints that took an array of params so we used an array to make the websocket handler send multiple events. Now I've made multiple dispatches explicit with `dispatchMultiple`.

File: ui/src/app/base/sagas/websockets.ts [hunk 2]
```diff
@@ -71,7 +73,13 @@ export const fileContextRequests = new Map<
 >();
 
 // A store of websocket endpoints that are being polled.
-const pollingRequests = new Set<WebSocketEndpoint>();
+const pollingRequests = new Set<PollRequestId>();
+
+const pollRequestId = (action: WebSocketAction | AnyAction): PollRequestId => {
+  const endpoint = `${action.meta?.model}.${action.meta?.method}`;
```
Comment (huwshimi) [inline ui/src/app/base/sagas/websockets.ts:79]: Polling used to be for a single endpoint as there's no point having multiple polls for the same request, however, for checking images we need to support multiple polling events as each check_images request includes a list of controller ids to check the images for.

File: ui/src/app/store/controller/selectors.test.ts [hunk 3]
```diff
@@ -100,4 +102,35 @@ describe("controller selectors", () => {
     });
     expect(controller.processing(state)).toStrictEqual(["abc123"]);
   });
+
+  it("can get the loaded state", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/store/controller/selectors.test.ts:106]: wrong test description

File: ui/src/app/store/controller/selectors.test.ts [hunk 4]
```diff
@@ -100,4 +102,35 @@ describe("controller selectors", () => {
...
+    });
+    expect(controller.imageSyncStatuses(state)).toStrictEqual(
+      controllerImageSyncStatusesFactory({
+        abc123: ImageSyncStatus.OutOfSync,
+        def456: ImageSyncStatus.Syncing,
+      })
+    );
+  });
+
+  it("can get a controller by id", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/store/controller/selectors.test.ts:123]: wrong test description

**Review by Caleb-Ellis — APPROVED** [review]
LGTM! Just some test descriptions need updating

**Pull request merged**